src/rebar3_hex.erl

```diff
@@ -51,6 +52,45 @@ task_args(State) ->
             {Task, proplists:delete(task, Opts)}
     end.
 
+command_state(State) ->
```

Comment:
Calling this `command_state` isn't exactly true, but isn't exactly wrong either. It really shouldn't be a big deal, but wanted thoughts on this so we can have consistent language going forward. If we look at what the user sees :   `rebar3 help [<namespace>] <task>`, however if I say `rebar3 help hex publish package`, then `package` gets recognized as the task.  I suppose we can `publish` is the default task, while `package` is a sub-task. 

So here I'm trying to make a clear distinction between the two saying `publish` is the command which may have a task argument. 

Alternatively, we can call this `task_state/1` and when we encounter a `task` tuple we could call that `command` or `sub_task` or something along those lines. What ever is going to be the least confusing for anyone who might hack on this and also in regards to documentation. 

Thoughts?

Reply:
I don't fully understand what you mean by `rebar3 help hex publish package` with `package` being the task; I sort of assume the task is publishing and that `package` is the type / an option / a parameter. I figure more specifically it specifies the handler or provider to run, which is more of an implementation detail.

I guess in general that's because `publish` is different from the other commands, where you use `mix hex.owner add|transfer|remove|...`, `mix hex.repo add|set|remove|list`, `mix hex.user auth|deauth|key <op>|...` where the general structure is `mix hex.<entity> <operation>` whereas publishing is clashing by being `mix hex.<operation> <entity>`. The `search` and `retire` commands similarly clash there.

if we line up with the original implementation, we can't necessarily hope to be clearer than they are. We might do a broader category as you said of `<task> <subtask>`, which might be as clear as we can get. Any arbitrary terms of this type should work, like `<category> <provider>`, `<group> <command>`, `<meta> <command>`, etc. As long as you carry in that meaning of "one container for the specific" you should be good to go.

Reply:
Ahh please ignore, I saw `{task, package}` ending up somewhere but that's because of something silly I was doing. I'm going to rename this to `task_state/1`